### DISCUSSION TITLE
SMTP use/Tracking etc

### DISCUSSION THREAD

**Discussion by EVCV** [mettle/sendportal#215] [General] [answered]
I want to use this initially just for my own company to sort out its mailer campaign as we currently use Mailchimp and it can get quite expensive when I am already running a hosting company so have email provisions with that.

My question is, I saw it does not work with SMTP for the tracking etc, Is there any way to integrate it so that i can use my own email servers but track all the data the same?

Also, could this be used to create an email marketing company? I have not looked into the installation yet as want to know the answer to these questions before doing so as then can work out my options.

**Comment by JonoB** [comment, marked as answer]
Yes, sendportal supports SMTP: https://github.com/mettle/sendportal-core/pull/112

**Reply by EVCV** [reply]
I saw somewhere (cant remember where in the files) that it says if SMTP is ues tracking of campaigns is not supported (stuff like deliveries, open, clicks etc) 

Is this still the case or is this now supported as that link shows it is, but then i found something that said it wasn't? my email on my hosting setup is roundcube.

**Comment by JonoB** [comment]
Open and click tracking is not supported on smtp. If you want that functionality, then select one of the other options (mailgun, postmark, etc)

**Comment by izdrail** [comment]
I'm the founder of Laravel Mail, SMTP notification it is posibble if the server support them using rpc comands 
I'n our platform we have implemented support for  smtp notification see 
```
class CustomEsmtpTransport extends EsmtpTransport
{
    public function executeCommand(string $command, array $codes): string
    {
        $command = match (true) {
            str_starts_with($command, 'MAIL FROM:') && isset($this->getCapabilities()['DSN']) => substr_replace($command, ' RET=HDRS', -2, 0),
            str_starts_with($command, 'RCPT TO:') && isset($this->getCapabilities()['DSN']) => substr_replace($command, ' NOTIFY=FAILURE', -2, 0),
            default => $command,
        };

        $response = parent::executeCommand($command, $codes);

        if (str_starts_with($command, 'EHLO ')) {
            $response .= "250 DSN\r\n";
        }

        return $response;
    }
}
```
